### ISSUE TITLE
[Bug]: PC0030 false positive when variable scope escapes function as return value

### ISSUE THREAD

**Issue by lv-janpieter** [ALCops/Analyzers#429]
### Analyzer

PlatformCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

PC0030

### Describe the bug
The `PC0030` analyzer ("Use partial records on read operation") incorrectly reports a warning/info message on record variables that escape the current function scope via an `exit()` statement. 

While the analyzer correctly ignores variables passed as `var` parameters, or when modifying/validating data, it fails to recognize that a record returned via `exit` leaves the local scope. Therefore, the analyzer should not enforce `SetLoadFields` here, as the calling procedure might require the full record context.

### To Reproduce
1. Enable ALCops with `PC0030` active.
2. Create a standard getter-procedure that retrieves a record locally and exits it.

**Code snippet:**
```al
internal procedure GetSalesHeader(OrderNo: Code[20]): Record "Sales Header"
var
    SalesHeader: Record "Sales Header";
begin
    // PC0030 is triggered here, but the record is escaping the function scope below
    SalesHeader.Get(SalesHeader."Document Type"::Order, OrderNo);
    exit(SalesHeader); 
end;
```

### Expected behavior
The analyzer should inspect the `ExitStatementSyntax` within the local block. If the local record variable is passed to an `exit()` statement, it escapes the local scope and `PC0030` should not be reported.

### Desktop Environment
- VS Code / Cursor
- AL Language Extension version: 17.0.2273547
- ALCops version: 1.3.4

**Comment by Arthurvdv** [comment]
I've published a new beta release, `v1.1.0-beta.2`, which should resolve this issue.

Could you verify for me if it's fixed on your end? Would be great to hear whether it's resolved or you're still running into problems.

If you're using the [ALCops VS Code extension](https://marketplace.visualstudio.com/items?itemName=Arthurvdv.alcops), you can switch to the beta channel in the extension settings. To force an update check, run the `ALCops: Check for Updates` command from the Command Palette.

**Comment by lv-janpieter** [comment]
Appreciate your effort. I verified the fix against several repositories that previously reported false positives, and all of them have disappeared. I also tested a few valid scenarios by removing SetLoadFields, and the analyzer still behaves as expected. I believe this change addresses the false positives without affecting valid detections. Thanks for working on this, and I’m happy to approve the change.

**Comment by Arthurvdv** [comment]
Thanks for verifying and great to hear it's resolved.

Closing this issue now that `v1.1.0` has been released.
